api/java/java_download.go

```diff
@@ -43,10 +45,36 @@ func InitialiseJavaEndpoints(r *mux.Router) {
 		if viper.GetBool("java.index.reindex") {
 			javaEndpoints.ReIndex()
 		}
+	}
+}
+
+// prepareProxies : Cycle through the submitted proxies and sanitize them.
+func prepareProxies() []*url.URL {
+
+	// Cycle through the repositories that are available.
+	configProxies := viper.GetStringSlice("java.proxies")
+	var proxies []*url.URL
+
+	if len(configProxies) > 0 {
+		for _, proxy := range configProxies {
+			url, err := url.ParseRequestURI(proxy)
+			if err != nil {
+				log.WithFields(log.Fields{"module": "api", "action": "InitialiseJavaEndpoints"}).Error(proxy + " is not a valid URI, ignoring...")
+			} else {
+				proxies = append(proxies, url)
+				log.WithFields(log.Fields{"module": "api", "action": "InitialiseJavaEndpoints"}).Info(url.String() + " configured as Java Proxy.")
+			}
+		}
+	}
 
-		// Set a default for the proxies - just in case they are not configured
-		viper.SetDefault("java.proxies", []string{"https://repo.maven.apache.org/maven2/"})
+	// If we've not managed to find any actual URLs to use as a proxy.
+	if len(proxies) == 0 {
```

Comment:
This could just be an else rather than evaluating again.

Reply:
Not necessarily. The proxies are being sanitized so even if there are several listed in the config they may not be valid and we end up with 0 anyway. So I'm not going to change this if that's ok?

Reply:
Yep, that makes perfect sense.